File: moviejava/app/src/main/java/com/example/moviejava/data/remote/movie/MovieService.java
package com.example.moviejava.data.remote.movie;


import com.example.moviejava.model.JsonObject;
import com.example.moviejava.model.up_coming.UpComing;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface MovieService {
    @GET("movie/top_rated")
    Observable<JsonObject> getTopRate(@Query("page") int page);

    @GET("movie/upcoming")
    Observable<UpComing> getUpcoming(@Query("page") int page);

    @GET("movie/popular")
    Observable<JsonObject> getPopular(@Query("page") int page);

    @GET("movie/now_playing")
    Observable<UpComing> getNowPlaying(@Query("page") int page);

    @GET("search/movie")
    Observable<JsonObject> searchMovie(@Query("query") String query, @Query("page") int page);


}

File: movie/app/src/main/java/com/example/movie/data/remote/movie/MovieService.kt
package com.example.movie.data.remote.movie

import com.example.movie.data.remote.Constant
import com.example.movie.model.JsonObject
import com.example.movie.model.up_coming.UpComing
import retrofit2.http.GET
import retrofit2.http.Query

interface MovieService {
    @GET("movie/top_rated")
    suspend fun getTopRate(@Query("page") page: Int): JsonObject

    @GET("movie/upcoming")
    suspend fun getUpcoming(@Query("page") page: Int): UpComing

    @GET("movie/popular")
    suspend fun getPopular(@Query("page") page: Int): JsonObject

    @GET("movie/now_playing")
    suspend fun getNowPlaying(@Query("page") page: Int): UpComing

    @GET("search/movie")
    suspend fun searchMovie(@Query("query") query: String, @Query("page") page: Int): JsonObject


}